TaskWithSmartPointers/MyUniquePtr.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+#include <iostream>
+
+template <class T>
+class MyUniquePtr
+{
+private:
+	T* m_ptr = nullptr;
+	friend std::ostream& operator<<(std::ostream& os, MyUniquePtr const& m);
+public:
+	T* get() const noexcept
+	{
+		return m_ptr;
+	}
+
+	T* release() noexcept
+	{
+		T* p = m_ptr;
+		m_ptr = nullptr;
+		return p;
+	}
+
+	void reset(T* p) noexcept
+	{
+		delete m_ptr;
+		m_ptr = p;
+	}
+
+	T& operator*()
+	{
+		return *(this->m_ptr);
+	}
+
+	MyUniquePtr() // default constructor
+	{
+		m_ptr = nullptr;
```

Comment:
m_ptr already has a default value in members.

Reply:
Done